shell/src/main/java/alluxio/cli/fs/command/CpCommand.java

```diff
@@ -61,6 +66,64 @@
           .hasArg(false)
           .desc("copy files in subdirectories recursively")
           .build();
+  private static final String COPY_PROGRESS_DONE = "#";
+
+  private ThreadPoolExecutor mCopyExecutor;
+  private BlockingQueue<String> mCopyProgress;
+  private Thread mCopyProgressDisplayThread;
+
+  /**
+   * Initializes necessary resources for async copy.
+   */
+  private void initAsyncCopy() {
+    final int numThreads = mFsContext.getConf().getInt(PropertyKey.USER_COPY_FROM_LOCAL_THREADS);
+    mCopyExecutor = new ThreadPoolExecutor(numThreads, numThreads,
+      1, TimeUnit.SECONDS, new ArrayBlockingQueue<>(numThreads * 2),
+      new ThreadPoolExecutor.CallerRunsPolicy());
+    mCopyProgress = new LinkedBlockingQueue<>();
+    mCopyProgressDisplayThread = new Thread(() -> {
+      while (true) {
```

Comment:
could use `!Thread.interrupted()` as the loop condition to be safe

Reply:
done